uploading bb synthetic control methods and associated images

VirginiaMirabile opened the pull request.

VirginiaMirabile committed:
uploading bb synthetic control methods and associated images

valerievossen commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
````diff
@@ -0,0 +1,207 @@
+```
+{{% /codeblock %}}
+
+The dataset we are going to use in this tutorial is the *smoking* dataset, based on the paper by Abadie et al. (2010), which evaluates the impact of tobacco control program on cigarette consumption in California.
````
Maybe you can add a link to Abadie et al. (2010)?

valerievossen commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
````diff
@@ -0,0 +1,207 @@
+  
+  smoking %>%
+  
+  # initiate the synthetic control object
+  synthetic_control(outcome = cigsale, # outcome
+                    unit = state, # unit index in the panel data
+                    time = year, # time index in the panel data
+                    i_unit = "California", # unit where the intervention occurred
+                    i_time = 1988, # time period when the intervention occurred
+                    generate_placebos=T # generate placebo synthetic controls (for inference)
+                    ) %>%
+  # Generate the aggregate predictors used to fit the weights (in our example referred to as W')
+  
+  # average log income, retail price of cigarettes, and proportion of the
+  # population between 15 and 24 years of age from 1980 - 1988
+  generate_predictor(time_window = 1980:1988,
+                     ln_income = mean(lnincome, na.rm = T),
+                     ret_price = mean(retprice, na.rm = T),
+                     youth = mean(age15to24, na.rm = T)) %>%
+  # average beer consumption in the donor pool from 1984 - 1988
+  generate_predictor(time_window = 1984:1988,
+                     beer_sales = mean(beer, na.rm = T)) %>%
+  
+  # Lagged cigarette sales 
+  generate_predictor(time_window = 1975,
+                     cigsale_1975 = cigsale) %>%
+  generate_predictor(time_window = 1980,
+                     cigsale_1980 = cigsale) %>%
+  generate_predictor(time_window = 1988,
+                     cigsale_1988 = cigsale) %>%
+  
+   # Generate the fitted weights for the synthetic control
+  generate_weights(optimization_window = 1970:1988, # time to use in the optimization task
+                   margin_ipop = .02,sigf_ipop = 7,bound_ipop = 6 # optimizer options
+  ) %>%
+  
+  # Generate the synthetic control
+  generate_control()
+```
+{{% /codeblock %}}
````
Is there a possibility to use more code blocks for this code? And maybe add the intuition in text-form instead of comments with #? I can't read the full comment without scrolling the big code block (maybe it's my laptop). I would keep the comments in the code with # as short as possible or otherwise divide them over two lines, so you don't have to scroll too much! 

And, is "aggregate predictor" the same as the weighting vector W? Maybe use the same word for that!

valerievossen commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
````diff
@@ -0,0 +1,207 @@
+```
+{{% /codeblock %}}
+
+Once the synthetic control is generated, one can easily assess the fit by comparing the trends of the synthetic and observed time series. The idea is that the trends in the pre-intervention period should map closely onto one another. Let's verify that with a graph: 
````
Maybe you could make a new section; two ways to check the validity of the synthetic control ? 1. compare pre-intervention trends. 2. compare observed covariates - just an idea :)

valerievossen commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
```diff
@@ -0,0 +1,207 @@
+There can be considerable variability in estimates depending on the specification of the synthetic control model.
+- Constructing synthetic controls can be computationally intensive, especially with large datasets or complex models.
+
+## Summary 
```
```suggestion

```

valerievossen commented:
Hi @VirginiaMirabile, 


I had a look over it. It reads super clear! I made a few comments that might be useful! And also something general I noticed: 
- Since the causal inference content is rearranged, there is a new section for Synthetic control, I would include it in there instead of the Introduction section!

- Also, the topic that now exists already (https://tilburgsciencehub.com/topics/analyze/causal-inference/synthetic-control/synth-control/) might be overlapping in some ways. We can maybe adjust this content so it's complementary (it's still useful as it contains Stata code as well), but adjust it as a follow-up on your article such that the content is not overlapping?

Let me know if something is unclear!

alexandervossen commented:
@VirginaMirabile did you include the feedback? Can I merge ? :)

srosh2000 commented:
@alexandervossen I need to review this PR too. Will do so by end of this week (caught up with bunch of grading deadlines)..

srosh2000 commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
```diff
@@ -0,0 +1,207 @@
+
+The method aims to generate a synthetic California using information from a subset of control states (the “donor pool”) where a similar law was not implemented. The donor pool is the subset of case comparisons from which information is borrowed to generate a synthetic version of the treated unit (“California”). 
+
+{{% codeblock %}}
```
I'd split the code chunks and take the reader through each step by step:

part 1: synthetic_control function code => explain what is the objective here, which function can be used(i.e. `synthetic_control`) and the options you specify

part 2: generate_predictor => here you are creating predictors from pre-intervention data. Again, motivate why we do this, which function we use and then code snippet. Be specific as to why for eg we take lagged cig sales ( -> to capture historical trends) and not just the code snippet. This will guide readers as to what kind of variables one could potentially use as pre trentment characteristics.

part 3: generate_weights => here we aim to find the optimal weights w_j for each control unit. explain the function and the arguments it takes clearly. 

part 4: generate_control()

srosh2000 commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
```diff
@@ -0,0 +1,207 @@
+ 
+## Advantages and limitations 
+
+According to Abadie (2021), the advantages of synthetic control methods are: 
```
Advantages seem a bit generic (esp point 1 and 3), some missing points:

- Sparsity: Since only a few control units are assigned non-zero weights, only a small subset out of a large pool of control units is used to construct the synthetic control. This prevents overfitting of the pre-intervention data, leading to more robust post-intervention inferences.

- No Extrapolation: Unlike some regression-based methods that may make predictions outside the range of the observed data, SCM ensures that the constructed synthetic control stays within the actual observerd outcomes of the control units. 

- Safeguard against specification searches and p-hacking: Synthetic control weights can be calculated and pre-registered before the post-treatment outcomes are realized, or before the actual intervention takes place.

srosh2000 commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
```diff
@@ -0,0 +1,207 @@
+- SCMs offer a transparent and data-driven method to construct the control group, which is based on observable characteristics and pre-treatment outcomes.
+- SCMs are flexible and can be applied in a wide range of contexts. 
+
+Possible disadvantages include: 
```
Missing disadvantages:

- Bias in small samples: In cases of small pre-treatment periods (T0), large donor pools(J), and substantial noise, there is a higher risk of interpolation bias(i.e. the synthetic control is overly tailored to fit the idiosyncracies of the pre-treatment period) and overfitting.

- Inference: Traditional methods like t-tests and p-values don't apply well. Permutation tests are used which have limitations, especially with small donor pools and the assumption that any control unit could have been treated ( which will be violated if control units are very different from each other or the treated unit). This kind of inference is also computationally demanding.

srosh2000 commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
```diff
@@ -0,0 +1,207 @@
+
+{{% summary %}}
+
+- Synthetic control methods (SCMs) are statistical techniques used to estimate the effect of interventions by creating a synthetic version of the treatment group from a weighted combination of control units.
```
include the functions used for estimation too and explain what they do. Write the function names under (``) -> e.g. `generate_predictor()`

srosh2000 commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
```diff
@@ -0,0 +1,207 @@
+{{% /summary %}}
+
+
+### References 
```
write each reference in bullet points and hyperlink. For eg:

- [Using Synthetic Controls: Feasibility, Data Requirements, and Methodological Aspects](https://www.aeaweb.org/articles?id=10.1257/jel.20191450 DOI: 10.1257/jel.20191450)

srosh2000 commented:
@VirginiaMirabile Added my comments :) As discussed feel free to pick new stuff from the older BB on SCM (e.g. STATA codes) and other impo points: https://tilburgsciencehub.com/topics/analyze/causal-inference/synthetic-control/synth-control/ 

@lachlandeer @alexandervossen we might have to remove the older version as this is more complete.

VirginiaMirabile committed:
implemented feddback + image

VirginiaMirabile commented:
Hi @srosh2000 @valerievossen, I have uploaded the revised building block version implementing your feedback. The main changes include:
- simplifying the maths, linking it to the use case of tobacco control program. 
- split and added more explanation to the code blocks. 
- improved advantages and disadvantages section. 

Let me know if any further work is needed!

valerievossen commented:
@VirginiaMirabile 
Looks good!

Can you still move the topic to the synthetic control section of causal inference, (it is now in the introduction section). Maybe you need to merge the master into this branch first to see the synthetic control section locally?

And some small comments to make the text easier to read:
- You can use these ` ` to highlight any "code" names in text, e.g.  `smoking` dataset, the `tidysynth` package, the time index (`year`), or  `plot_differences()`
- You can use italics to start the bullet points (in last section about advantages and disadvantages) for example - *Bias in small samples:* ...

srosh2000 commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
```diff
@@ -0,0 +1,268 @@
+
+In this step, we create the synthetic control object using the synthetic_control function. We specify the outcome variable (cigsale), the unit index (state), and the time index (year). We also identify the intervention unit (California) and the intervention time (1988). The *generate_placebos* option is set to TRUE to allow for placebo controls.
+
+Next, we generate predictors for the model. Predictors are the variables used to characterize each unit in the dataset before the intervention, they are the features or attributes based on which the similarity between the treated unit and control units is assessed. The synthetic control method uses these predictors to ensure that the synthetic control unit closely matches the treated unit in the  ...
```
would help to have subheadings for each step? name it as :

- Step 1: Initiate synthetic control object
- Step 2: Generate predictors 

and so on..

srosh2000 commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
```diff
@@ -0,0 +1,268 @@
+
+According to Abadie (2021), the advantages of synthetic control methods are: 
+
+- Replicability: SCMs offer a transparent and data-driven method to construct the control group, which is based on observable characteristics and pre-treatment outcomes.
```
would be good to have the bullet point header in bold. E.g. **Replicability**

srosh2000 commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
````diff
@@ -0,0 +1,268 @@
+```
+{{% /codeblock %}}
+ 
+## Advantages and limitations 
````
have separate subheading for "Advantages" and later "Disadvantages"

srosh2000 commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
```diff
@@ -0,0 +1,268 @@
+
+- Synthetic control methods (SCMs) are statistical techniques used to estimate the effect of interventions by creating a synthetic version of the treatment group from a weighted combination of control units.
+- SCMs provide a transparent, data-driven approach for constructing control groups, making them suitable for contexts where randomized controlled trials are not feasible.
+- While SCMs offer flexibility and systematic estimation, they require extensive pre-treatment data and can be sensitive to model specifications and computationally intensive.
```
include points about which function can be used for implementation.

srosh2000 commented on content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md:
```diff
@@ -0,0 +1,268 @@
+Control Program](https://www.tandfonline.com/doi/abs/10.1198/jasa.2009.ap08746), Journal of the American Statistical Association, 105:490, 493-505, DOI:
+10.1198/jasa.2009.ap08746
+
+- [Code snippets](https://github.com/edunford/tidysynth/blob/master/README.md)
```
rename as [Tidysynth: A Tidy Implementation of Synthetic Control Methods]( add link)

srosh2000 reviewed (commented):
Content looks good :)
Have included some comments related to formatting/structuring to polish things. Thanks a lot!

shrabasteebanerjee commented:
Hi @VirginiaMirabile , let us know when you are ready with these changes!

VirginiaMirabile committed:
Implementation of final feedback

VirginiaMirabile commented:
Hi all, I have addressed all the latest feedback. For some reason, the folder of causal inference doesn't show up for me so I can't move it just yet. I will try to figure it out, but if you know why that might be let me know!

alexandervossen commented:
Hi @srosh2000  and @shrabasteebanerjee - do you think we can merge this?

shrabasteebanerjee commented:
hi @alexandervossen i think this one we can discard and merge the other pull request from Aug 29th
